datafusion/sql/src/planner.rs

```diff
@@ -498,7 +498,30 @@ impl<'a, S: ContextProvider> SqlToRel<'a, S> {
             SQLDataType::Date => Ok(DataType::Date32),
             SQLDataType::Time => Ok(DataType::Time64(TimeUnit::Nanosecond)),
             SQLDataType::Timestamp => Ok(DataType::Timestamp(TimeUnit::Nanosecond, None)),
-            _ => Err(DataFusionError::NotImplemented(format!(
+            // Explicitly list all other types so that if sqlparser
```

Comment:
This does not change the behavior, but makes the current behavior more explicit in my opinion

It also makes clear some surprising things (like  several `unsigned` variants appear not to be supported along with `interval` as noted by @waitingkuo  in https://github.com/apache/arrow-datafusion/issues/3166#issuecomment-1215832724